### PR TITLE
PyPI release & Readme

### PR DESCRIPTION
**Pull request by breznak**
- [x] fix pip install on linux
- [x] improve Dockerfile for ARM64
- [x] arm builds now part of whl/releases
- [x] updated README

Fixes: #19 
Fixes: #361

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,11 +25,18 @@ jobs:
...
 
-    - name: Set up Python ${{ matrix.python-version }}
+    - name: Set up Python ${{ matrix.python-version }} for PR
+      if: github.event_name != 'push'
       uses: actions/setup-python@v1
       with:
         python-version: ${{ matrix.python-version }}
 
+    - name: Set up old Python for PyPI
+      if: github.event_name == 'push'
```
Comment (breznak) [inline .github/workflows/build.yml]: need to use python 3.4 during Release on linux for compatible "manylinux" PyPI (user's python must be newer/same)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -128,6 +134,12 @@ jobs:
         name: dist-windows-2019
         path: dist/
 
+    - uses: actions/download-artifact@master
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+      with:
+        name: dist-arm64
```
Comment (breznak) [inline .github/workflows/build.yml]: PyPI now includes arm64 whl

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -172,3 +184,57 @@ jobs:
       run: |
         cd build/scripts
         ../Debug/bin/unit_tests
+
+
+
+  build-arm64-docker:
+    name: Build for ARM64 on Docker
```
Comment (breznak) [inline .github/workflows/build.yml]: merged the ARM build into this file, for easier integration to PYPI,Release jobs

File: Dockerfile [hunk 4]
```diff
@@ -2,19 +2,31 @@
...
 #   Our circleci arm64 build uses this specifically.
 #   https://docs.docker.com/engine/reference/commandline/build/
-ARG arch=amd64
+#target compile arch
+ARG arch=arm64
+#host HW arch
+ARG host=amd64
 
+## Stage 0: deboostrap: setup cross-compile env 
+FROM multiarch/qemu-user-static as bootstrap
```
Comment (breznak) [inline Dockerfile]: docker now automatically resets the platform (so CI, user do not have to do that), allowing easy host-arch cross-compilation. 

Docker also creates whl, and tar.gz releases for PyPI and GH Releases

File: README.md [hunk 5]
```diff
@@ -2,9 +2,7 @@
 
 # htm.core
 
-[![Linux/OSX Build Status](https://travis-ci.org/htm-community/htm.core.svg?branch=master)](https://travis-ci.org/htm-community/htm.core)
-[![OSX CircleCI](https://circleci.com/gh/htm-community/htm.core/tree/master.svg?style=svg)](https://circleci.com/gh/htm-community/htm.core/tree/master)
-[![Windows Build status](https://ci.appveyor.com/api/projects/status/59f87and1x0ugss9/branch/master?svg=true)](https://ci.appveyor.com/project/htm-community/nupic-cpp/branch/master)
+[![CI Build Status](https://github.com/htm-community/htm.core/workflows/build/badge.svg)](https://github.com/htm-community/htm.core/actions)
```
Comment (breznak) [inline README.md]: fixed badges after CI switch

**Comment by breznak** [comment]
Merged to test with tag, please post-review @dkeeney when you have time

File: Dockerfile [hunk 6]
```diff
@@ -24,21 +36,36 @@ ADD . /usr/local/src/htm.core
...
         -r requirements.txt
-RUN python3 setup.py install --force
+RUN python setup.py install --force
 
 # Test
-RUN python3 setup.py test
+RUN python setup.py test #Note, if you get weird import errors here, 
+# do `git clean -xdf` in your host system, and rerun the docker
+
+## Stage 2: create release packages (for PyPI, GH Releases)
```
Comment (dkeeney) [inline Dockerfile]: Do users need to know how to build wheels?
I guess it is a good thing to document how to do it for us maintainers.

File: README.md [hunk 7]
```diff
@@ -38,18 +36,26 @@ in C++ library.
...
 
-# Installation
+## Installation
 
-## Binary releases
+### Binary releases
+
+If you want to use `htm.core` from Python, the easiest method is to install from [PyPI](https://test.pypi.org/project/htm.core/)
+```
+python -m pip install -i https://test.pypi.org/simple/ htm.core
```
Comment (dkeeney) [inline README.md]: Don't we have our wheels in pypi.org yet?
I don't think the general public can access test.pypi.org.

File: README.md [hunk 8]
```diff
@@ -228,28 +239,27 @@ not include -DBINDING_BUILD=Python3 in the CMake command.
...
-obtain one and link with it, but a way to better isolate the python extension is to build a special main( )
-as explained here: https://pythonextensionpatterns.readthedocs.io/en/latest/debugging/debug_in_ide.html.
+The problem is that it requires a debug version of the python library, `python37_d.lib`.  It is possible to
+obtain one and link with it, but a way to better isolate the python extension is to build a special `main( )`
+as [explained in debugging Python](https://pythonextensionpatterns.readthedocs.io/en/latest/debugging/debug_in_ide.html).
 
 Be aware that the CMake maintains a cache of build-time arguments and it will ignore some arguments passed
 to CMake if is already in the cache.  So, between runs you need to clear the cache or even better,
-entirely remove the build/ folder.
+entirely remove the `build/` folder (ie. `git clean -xdf`).
```
Comment (dkeeney) [inline README.md]: Actually I use `rm -r build`.  
I did not know that `git clean ` existed.  I will have to give it a try.

File: Dockerfile (hunk 6 above)
Comment (breznak) [reply Dockerfile]: > Do users need to know how to build wheels? 

I don't think so (?) 
And we also do it the standard way, `python setup.py bdist_wheel`

File: README.md (hunk 7 above)
Comment (breznak) [reply README.md]: no, not yet. 
And there's not too much pressure (atleast until we iron all the bugs in wheels regarding compatibility), yes, anyone can install using the command above. Actually, it's now successfully used in community/NAB repo.

**Pull request merged**